server/util/perms/perms.go

```diff
@@ -46,7 +46,15 @@ func AnonymousUserPermissions() *UserGroupPerm {
 	}
 }
 
-func GroupAuthPermissions(groupID string) *UserGroupPerm {
+func GroupAuthPermissions(u interfaces.UserInfo) *UserGroupPerm {
+	return &UserGroupPerm{
+		UserID:  u.GetUserID(),
+		GroupID: u.GetGroupID(),
+		Perms:   GROUP_READ | GROUP_WRITE,
+	}
+}
+
+func GroupAuthPermissionsByGroupID(groupID string) *UserGroupPerm {
```

Comment:
maybe call this something like `GroupPermissionsOnly` since only the group is given permissions, not the user.

(Side note, I think it was a mistake to set `UserID` to `groupID` here - a group is not a user so this doesn't "typecheck" if that makes sense. Not sure if anything would break if we changed it so probably just leave as-is :shrug: )

Reply:
+1

Reply:
I renamed this function to be GroupPermissionsDeprecated. I added a comment to the function; basically this is only used when we insert into workflow table. It requires some refactoring to make it use DefaultPermissions. Since that table is going to be replaced by GitRepositories in the future, I think we can just keep the behavior unchanged.